atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/persistentlock/CheckAndSetExceptionMapper.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.persistentlock;
+
+import java.util.List;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Iterables;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+
+public class CheckAndSetExceptionMapper implements ExceptionMapper<CheckAndSetException> {
```

Comment:
I thought it was only used when registered with `env.register`? In which case it would only be those CASExceptions under the umbrella of the persistent lock calls.